samples/nrf_desktop/src/modules/hid_state.c

```diff
@@ -648,17 +645,21 @@ static void report_issued(const void *subscriber_id, enum target_report tr,
 		return;
 	}
 
+	struct report_state *rs = &subscriber->state[tr];
+
+	__ASSERT_NO_MSG(rs->cnt > 0);
+	rs->cnt--;
+
 	if (state.selected != subscriber) {
 		LOG_INF("Subscriber %p not active", subscriber_id);
```

Comment:
This was the source of the bug. When we change the selected subscriber it does not mean that communication to the other subscriber cease. We don't want to send any new reports there but we need to gracefully finish the pending communication.
The change is doing just that - for disconnected subscribers we simply decrement the number of reports and change the state to IDLE if there are none pending.